internal/provider/provider.go

```diff
@@ -47,6 +48,11 @@ func (p *ProwlarrProvider) Schema(_ context.Context, _ provider.SchemaRequest, r
 				Optional:            true,
 				Sensitive:           true,
 			},
+			"authorization": schema.StringAttribute{
+				MarkdownDescription: "Token for token-based authentication with Prowlarr. This is an alternative to using an API key. Set this via the `PROWLARR_AUTHORIZATION` environment variable. Either `authorization` or `api_key` must be provided, but not both.",
```

Comment:
The `but not both` is not enforced anywhere.

Reply:
```go
	if key == "" && authorization == "" {
		resp.Diagnostics.AddError(
			"Missing Authentication Credentials",
			"Both 'api_key' and 'authorization' are empty. You must provide either an API key or an authorization token for Prowlarr authentication.",
		)
		return
	} else if key != "" && authorization != "" {
		resp.Diagnostics.AddError(
			"Conflicting Authentication Credentials",
			"Both 'api_key' and 'authorization' are provided. You must only provide one of these for Prowlarr authentication.",
		)
		return
	}
```
Looks good?

Reply:
Think so, but you can drop the `else if` part. A second, independent `if` statement is sufficient.